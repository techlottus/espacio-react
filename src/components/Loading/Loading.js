import React, { useEffect } from "react";
import { useSelector } from "react-redux";
import "./Loading.scss";

const Loading = () => {
  const loadingStore = useSelector((state) => state.loading);

  useEffect(() => {
    if (loadingStore.isShow) {
      document.body.style.overflowY = "hidden";
    } else {
      document.body.style.overflowY = "auto";
    }
  }, [loadingStore.isShow]);

  return (
    <div
      className="loading-container"
      style={{ visibility: loadingStore.isShow ? "visible" : "hidden" }}
    >
      <div className="loadingcontent">
        <div className="loading"></div>
      </div>
    </div>
  );
};

export default Loading;
