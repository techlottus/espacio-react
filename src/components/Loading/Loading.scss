@import '../../App.scss';

.loading-container {
    position: fixed;
    z-index: 100;
    width: 100vw;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: map-get($palette-overlay, b-50); 
    top: 0;

.loadingcontent {
    
    width: pixeltorem(16px);
    height: pixeltorem(16px);
    position: relative;

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
        width: pixeltorem(36px);
        height: pixeltorem(36px);
      }

    .loading {
        border-radius: 50%;
        position: absolute;
        width: 100%;
        height: 100%;
        border: pixeltorem(8px) solid map-get($palette-neutral, n10);
        border-top: pixeltorem(8px) solid palette-colors($brand, p00);
        animation: rotate 1s ease infinite;
        top: pixeltorem(10px);
        left: pixeltorem(10px);
    }

    @keyframes rotate {
        from {transform: rotate(0);}
        to {transform: rotate(360deg);}
        
    }
}

}
