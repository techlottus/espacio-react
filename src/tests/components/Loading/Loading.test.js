import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import Loading from "../../../components/Loading/Loading";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState = {
  loading: { isShow: true },
};

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <Loading />
    </Provider>
  );
};

describe("Test <Loading/>", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  test("should hidden false", () => {
    initState = { loading: { isShow: false } };
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
