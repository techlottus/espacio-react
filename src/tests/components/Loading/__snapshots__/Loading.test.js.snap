// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Loading/>  1`] = `
<div
  className="loading-container"
  style={
    Object {
      "visibility": "hidden",
    }
  }
>
  <div
    className="loadingcontent"
  >
    <div
      className="loading"
    />
  </div>
</div>
`;

exports[`Test <Loading/> should hidden false 1`] = `
<div
  className="loading-container"
  style={
    Object {
      "visibility": "hidden",
    }
  }
>
  <div
    className="loadingcontent"
  >
    <div
      className="loading"
    />
  </div>
</div>
`;

exports[`Test <Loading/> should init component 1`] = `
<div
  className="loading-container"
  style={
    Object {
      "visibility": "visible",
    }
  }
>
  <div
    className="loadingcontent"
  >
    <div
      className="loading"
    />
  </div>
</div>
`;
